Refactored asset reference and created asset guide

mercul3s opened the pull request:
Closes #264 
Closes #194

palourde commented:
:+1:

calebhailey commented:
Let's not duplicate documentation for underlying Uchiwa features. We have omitted this because it's not recommended or supported for Sensu Enterprise customers.

mercul3s committed:
Refactored asset reference and created asset guide

palourde commented on content/sensu-core/2.0/guides/install-check-executables-with-assets.md:
```diff
@@ -0,0 +1,61 @@
+---
+title: "How to install Check Executables using Assets"
+linkTitle: "Install Check Executables Using Assets"
```
We might want to use a shorter title:

![screen shot 2018-04-06 at 1 45 33 pm](https://user-images.githubusercontent.com/4105580/38436082-acaba060-39a1-11e8-975d-ec5c4d242dd4.png)

mercul3s replied:
Ah yeah, I will think of something shorter!

mercul3s replied:
@palourde updated!

palourde commented on content/sensu-core/2.0/guides/install-check-executables-with-assets.md:
```diff
@@ -0,0 +1,61 @@
+sensuctl asset create check_website.tar.gz \
+  -u http://example.com/check_website.tar.gz \
+  --sha512 "$(sha512sum check_website.tar.gz | cut -f1 -d ' ')"
+OK
```
We can probably remove that bit :)

palourde commented on content/sensu-core/2.0/guides/install-check-executables-with-assets.md:
```diff
@@ -0,0 +1,61 @@
+{{< highlight shell >}}
+sensuctl asset create check_website.tar.gz \
+  -u http://example.com/check_website.tar.gz \
+  --sha512 "$(sha512sum check_website.tar.gz | cut -f1 -d ' ')"
```
Looks like `sha512sum` is not installed by default on macOS; we should probably figure out a command that will work on _most_ systems!

portertech replied:
I use `shasum -a 512` on my mac.

mercul3s replied:
Yeah, I was wondering about that too - it's just shasum on macOS. I looked to see if I could brew install sha512sum and there isn't a package for it. I can be a little more verbose here to make it clearer.

palourde commented on content/sensu-core/2.0/guides/install-check-executables-with-assets.md:
```diff
@@ -0,0 +1,61 @@
+{{< /highlight >}}
+
+This command will set a check's assets to `check_website.tar.gz`.
+
```
Maybe we should add a small section explaining how to validate if the asset properly worked. Example: https://docs.sensu.io/sensu-core/2.0/guides/send-slack-alerts/#validating-the-handler

palourde commented on content/sensu-core/2.0/reference/assets.md:
```diff
@@ -1,92 +1,101 @@
 version: "2.0"
 product: "Sensu Core"
-platformContent: true
+platformContent: false 
```
👍 👍 👍

palourde commented on content/sensu-core/2.0/reference/assets.md:
```diff
@@ -1,92 +1,101 @@
-This command will set a check's assets to `readme` and `foobar`.
+## How do assets work?
+At runtime, the agent sequentially fetches assets and stores them in its local
+cache. Asset dependencies are then injected into the `PATH` so they are
```
Do you think it would be useful to reference the `--cache-dir` flag of `sensu-agent` when referring to the agent local cache?

mercul3s replied:
Oh yeah, good point. I'll make a note of that.

palourde commented on content/sensu-core/2.0/reference/assets.md:
```diff
@@ -1,92 +1,101 @@
+
+### Attributes
+
+name         | 
```
I've used the following format, inherited from 1.x on other reference docs: https://docs.sensu.io/sensu-core/2.0/reference/checks/#check-naming

But I think I like your approach better. Could you just add the link to the online regex validator? And we'll switch the rest of the docs to this format!

mercul3s replied:
I didn't even realize there was a link to that in other places - I'll add it!

palourde reviewed (changes requested):
Just some small details, otherwise amazing work 🎉 💯 👍

mercul3s committed:
Updated link title to be shorter

mercul3s committed:
Addressing PR feedback

Add alternative sha generation command
Mention sensu-agent cache dir
Add note about validating the check works with the asset

mercul3s commented:
@palourde ok I think I've addressed all the comments, thanks for the thorough review!

mercul3s committed:
Add alternative checksum generation

palourde commented on content/sensu-core/2.0/guides/install-check-executables-with-assets.md:
```diff
@@ -0,0 +1,78 @@
+  --sha512 "$(sha512sum check_website.tar.gz | cut -f1 -d ' ')"
+{{< /highlight >}}
+
+If you're using a mac, you'll need to use `$(shasum check_website.tar.gz | cut -f1 -d ' ') to generate a checksum.
```
Actually it should be `shasum -a 512` otherwise it uses the default algorithm, which is SHA-1 I believe.

mercul3s replied:
oh you are correct, I knew this and I still missed it lol :) fixing now!

palourde reviewed (approved):
Just a small thing about shasum for macOS, otherwise 👍 
![](https://user-images.githubusercontent.com/4105580/38505187-92ff6d32-3be4-11e8-935b-20c43933be3d.gif)

mercul3s committed:
Document correct shasum incantation
